### PR TITLE
Add invalid json exception handler

### PR DESCRIPTION
**Pull request by jlisam**
This PR adds an `ExceptionHandler` method that catches and handles `InvalidProtocolBufferException` (parsing invalid JSON) and returns a 400 status code along with a message.

### CODE REVIEW

File: runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java [hunk 1]
```diff
@@ -11,54 +15,67 @@
...
+      InvalidProtocolBufferException e) {
+
+    LOGGER.error("InvalidProtocolBufferException: request = {}, method = {}, msg= {}",
+        request.getRequestURI(), request.getMethod(), e.getMessage(), e);
 
-    @ExceptionHandler(FlitException.class)
-    public ResponseEntity<?> handleFlitException(HttpServletRequest request, FlitException e) {
+    Map<String, Object> response = new HashMap<>();
+    response.put("code", ErrorCode.INVALID_ARGUMENT);
+    response.put("msg", "Invalid JSON");
```
Comment (BenEddy) [inline runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java]: I think we'll want to preserve the error message because it's pretty helpful for debugging malformed requests (e.g. "Cannot find field: queues in message aqueduct.api.v1.SendRequest"), especially since `application/protobuf` or `application/json` can both throw `InvalidProtocolBufferException`, so "Invalid JSON" might not be accurate.

File: runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java [hunk 2]
```diff
@@ -11,54 +15,67 @@
...
 
-import javax.servlet.http.HttpServletRequest;
-import java.util.HashMap;
-import java.util.Map;
-
 @ControllerAdvice
 @Component
 public class FlitExceptionHandler extends ResponseEntityExceptionHandler {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(FlitExceptionHandler.class);
```
Comment (BenEddy) [inline runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java]: As much as I prefer 2 spaces for indentation, can we revert these indentation changes and respect the original project indentation conventions? 😄

File: plugin/src/main/shell/protoc-gen-flit [hunk 3]
```diff
@@ -15,7 +15,7 @@
 # The quiet flag is required otherwise the tooling will dump output to stdout which will break the
 # plugin output and generation will fail.
 # -----------------------------------------------------------------------------
-DIR=$(dirname $(readlink -f "$0"))
```
Comment (BenEddy) [inline plugin/src/main/shell/protoc-gen-flit]: I'd suggest removing this unrelated change and PRing it upstream separately.

File: runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java (hunk 1 above)
Comment (jlisam) [reply runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java]: sounds good!

File: runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java (hunk 2 above)
Comment (jlisam) [reply runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java]: sorry,  IntelliJ automatically formatted it. Let me fix that back

File: plugin/src/main/shell/protoc-gen-flit (hunk 3 above)
Comment (jlisam) [reply plugin/src/main/shell/protoc-gen-flit]: will do. I was trying to get the test to pass and without changing this, it fails (assumes that it's Linux).
